skbase/lookup/tests/test_lookup.py

```diff
@@ -0,0 +1,899 @@
+"""Tests for skbase lookup functionality.
+
+tests in this module:
+
+    test_is_non_public_module  - tests _is_non_public_module logic
+"""
+import importlib
+import pathlib
+from types import ModuleType
+from typing import List
+
+import pandas as pd
+import pytest
+
+from skbase.base import BaseEstimator, BaseObject
+from skbase.base._base import TagAliaserMixin
+from skbase.lookup import all_objects, get_package_metadata
+from skbase.lookup._lookup import (
+    _determine_module_path,
+    _filter_by_class,
+    _filter_by_tags,
+    _get_return_tags,
+    _import_module,
+    _is_ignored_module,
+    _is_non_public_module,
+    _walk,
+)
+from skbase.mock_package.mock_package import CompositionDummy, NotABaseObject
+from skbase.tests.conftest import FixtureClassParent
+
+__author__: List[str] = ["RNKuhns"]
+__all__: List[str] = []
+
+
+MODULE_METADATA_EXPECTED_KEYS = (
+    "path",
+    "name",
+    "classes",
+    "functions",
+    "__all__",
+    "authors",
+    "is_package",
+    "contains_concrete_class_implementations",
+    "contains_base_classes",
+    "contains_base_objects",
+)
+MOD_NAMES = {
+    "public": ("skbase", "skbase.lookup", "some_module.some_sub_module"),
+    "non_public": (
+        "skbase.lookup._lookup",
+        "some_module._some_non_public_sub_module",
+        "_skbase",
+    ),
+}
+REQUIRED_CLASS_METADATA_KEYS = [
+    "klass",
+    "name",
+    "description",
+    "tags",
+    "is_concreate_implementation",
+    "is_base_class",
+    "is_base_object",
+    "authors",
+    "module_name",
+]
+REQUIRED_FUNCTION_METADATA_KEYS = ["func", "name", "description", "module_name"]
+
+
+@pytest.fixture
+def fixture_exclude_classes_skbase_metadata_tests():
+    """Fixture of classes to exclude from tests of get_package_metadata on skbase."""
+    return TagAliaserMixin
+
+
+@pytest.fixture
+def mod_names():
+    """Pytest fixture to return module names for tests."""
+    return MOD_NAMES
+
+
+@pytest.fixture
+def fixture_object():
+    """Pytest fixture of BaseObject class."""
+    return BaseObject
+
+
+@pytest.fixture
+def fixture_not_a_base_object():
+    """Pytest fixture for NotABaseObject."""
+    return NotABaseObject
+
+
+@pytest.fixture
+def fixture_composition_dummy():
+    """Pytest fixture for CompositionDummy."""
+    return CompositionDummy
+
+
+@pytest.fixture
+def fixture_class_parent():
+    """Pytest fixture for FixtureClassParent."""
+    return FixtureClassParent
+
+
+@pytest.fixture
+def fixture_test_lookup_mod_path():
+    """Fixture path to the lookup module determined from this file's path."""
+    return pathlib.Path(__file__).parent.parent
+
+
+@pytest.fixture
+def fixture_skbase_root_path(fixture_test_lookup_mod_path):
+    """Fixture to root path of skbase package."""
+    return fixture_test_lookup_mod_path.parent
+
+
+def _check_package_metadata_result(results):
+    """Check output of get_package_metadata is expected type."""
```

Comment:
this function is hard to read.

* do you want to turn all instances of `result_ok =False; break (then stuff happens)` into a simple `return False`?
* the `if`/`elif` is also a bit confusing, as it implies a "switch" - could simply be a lot of "`if ... return False`"?

Reply:
I've made the suggested changes. Agreed it isn't super easy to read, hopefully that helps.